### DISCUSSION TITLE
Set Weights dynamically for the inference session

### DISCUSSION THREAD

**Discussion by fedemarche** [microsoft/onnxruntime#13642] [Other Q&A]
Hi.
I have a question regarding the InferenceSession. 

I was wondering if it is possible to maintain the same InferenceSession instantiation and switch the underlying set of weights of the Model given an equal one but with just different kernel weights. In order so to dynamically switch weights at run time.

Is it possible currently?
There is any workaround?
is it possible at all?
 
Thank you

**Comment by glaucouri** [comment]
We also would have great benefit if was possible to change at runtime as we do using set_weight of Keras models.
